### PR TITLE
[MAINT] update constructors of estimators in nilearn.connectome for sklearn compliance

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to https://github.com/nilearn/nilearn/issues/4538#issuecomment-2527259857

Changes proposed in this pull request:

- remove logic from constructor
- adds test for covarariance estimators

### CODE REVIEW

File: nilearn/connectome/connectivity_matrices.py [hunk 1]
```diff
@@ -370,7 +370,7 @@ def prec_to_partial(precision):
 
 
 @fill_doc
-class ConnectivityMeasure(BaseEstimator, TransformerMixin):
```
Comment (Remi-Gau) [inline nilearn/connectome/connectivity_matrices.py:373]: change required for proper methods order resolution

File: nilearn/connectome/tests/test_group_sparse_cov.py [hunk 2]
```diff
@@ -35,55 +35,11 @@ def test_check_estimator_group_sparse_covariance_cv(estimator, check, name):  #
 @pytest.mark.parametrize(
     "estimator, check, name",
     check_estimator(
-        estimator=[GroupSparseCovarianceCV()],
+        estimator=[GroupSparseCovarianceCV(), GroupSparseCovariance()],
         valid=False,
         extra_valid_checks=extra_valid_checks,
     ),
 )
-def test_check_estimator_invalid_group_sparse_covariance_cv(
```
Comment (Remi-Gau) [inline nilearn/connectome/tests/test_group_sparse_cov.py:43]: GroupSparseCovariance and GroupSparseCovarianceCV now validated the same checks so we can use a single parametrized test for both

File: nilearn/connectome/tests/test_connectivity_matrices.py [hunk 3]
```diff
@@ -36,6 +40,64 @@
 N_SUBJECTS = 5
 
 
+@pytest.mark.parametrize(
+    "estimator",
+    [EmpiricalCovariance(), LedoitWolf()],
```
Comment (Remi-Gau) [inline nilearn/connectome/tests/test_connectivity_matrices.py:45]: adds test for covarariance estimators

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Review by man-shu — APPROVED** [review]
LGTM! The codecov complaints seem to be unrelated?

**Comment by Remi-Gau** [comment]
> LGTM! The codecov complaints seem to be unrelated?

yeah I do not see a specific line missing coverage in the diff view

**Pull request merged**